### ISSUE TITLE
Error - unused argument (sample_reconstruction_plots = sample_reconstruction_plots)

### ISSUE THREAD

**Issue by eltonjrv** [AlexandrovLab/SigProfilerAssignmentR#2]
Dear SigProfilerAssignmentR developers,

Thanks for the tool!

I've successfully (I believe) gone through all your installation steps.
When attempting to run cosmic_fit I'm getting stuck with the following error:

> cosmic_fit(samples="./", output="./", input_type='vcf', context_type="96", collapse_to_SBS96=TRUE, cosmic_version=3.3, exome=FALSE, genome_build="GRCh38")
Error in Analyze$cosmic_fit(samples = samples, output = output, signatures = signatures,  : 
  unused argument (sample_reconstruction_plots = sample_reconstruction_plots)

Below is my python config:
> library(reticulate)
> py_config()
python:         /nobackup/fbsev/bioinformatics-tools/miniconda3/envs/SigProfAssignR/bin/python3
libpython:      /nobackup/fbsev/bioinformatics-tools/miniconda3/envs/SigProfAssignR/lib/libpython3.7m.so
pythonhome:     /nobackup/fbsev/bioinformatics-tools/miniconda3/envs/SigProfAssignR:/nobackup/fbsev/bioinformatics-tools/miniconda3/envs/SigProfAssignR
version:        3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 06:08:21)  [GCC 9.4.0]
numpy:          /nobackup/fbsev/bioinformatics-tools/miniconda3/envs/SigProfAssignR/lib/python3.7/site-packages/numpy
numpy_version:  1.21.6
sys:            [builtin module]

Any clue on that? 
I'm not even typing the "sample_reconstruction_plots" parameter from cosmic_fit function.

Thanks

**Comment by marcos-diazg** [comment]
Dear @eltonjrv,

Thanks so much for your interest in our tool. Could you please check which version of the Python package for SigProfilerAssignment you are using? You can do that by typing `pip list` from your terminal.

Thanks! Best,

Marcos

**Comment by eltonjrv** [comment]
Thanks for your prompt reply, Marcos.
Here it goes:
$ pip list
Package                    Version
-------------------------- ----------
about-time                 4.2.1
alive-progress             3.1.5
argcomplete                1.12.3
biopython                  1.76
chardet                    5.2.0
charset-normalizer         2.0.12
cryptography               36.0.2
cycler                     0.11.0
dnaio                      0.5.0
dxpy                       0.337.0
eggnog-mapper              2.1.12
grapheme                   0.6.0
joblib                     1.3.2
kiwisolver                 1.4.5
matplotlib                 3.4.3
numpy                      1.21.6
packaging                  23.2
pandas                     1.3.5
patsy                      0.5.6
Pillow                     9.5.0
pip                        24.0
psutil                     5.9.4
pyparsing                  3.1.1
PyPDF2                     3.0.1
python-dateutil            2.8.2
pytz                       2024.1
reportlab                  4.1.0
requests                   2.27.1
scikit-learn               1.0.2
scipy                      1.7.3
seaborn                    0.12.2
setuptools                 69.0.3
SigProfilerAssignment      0.0.19
SigProfilerMatrixGenerator 1.2.13
sigProfilerPlotting        1.3.2
six                        1.16.0
sklearn                    0.0.post12
statsmodels                0.13.5
threadpoolctl              3.1.0
typing_extensions          4.7.1
websocket-client           0.54.0
wheel                      0.42.0
XlsxWriter                 1.4.3
xopen                      1.0.1

Thanks again

**Comment by marcos-diazg** [comment]
Thanks for sending this! Could you please upgrade your SigProfilerAssignment Python package? You can do that by running `pip install SigProfilerAssignment --upgrade`. You are currently using a pretty old version (v0.0.19, from November 2022). Updating to the most recent v0.1.3 should solve your issue, but please let us know if that's not the case.

**Comment by eltonjrv** [comment]
Hi Marcus,

I've run the upgrade command you told me, but it maintains v0.0.19.
Attached is a log file from that upgrade command.
Any other tip to get it upgraded?

Thanks
[upgrade.log](https://github.com/AlexandrovLab/SigProfilerAssignmentR/files/14316910/upgrade.log)

**Comment by marcos-diazg** [comment]
Hi again Elton,

Unfortunately, I think the version of Python/Matplotlib installed in your `SigProfAssignR` conda environment is too old for the newest SigProfilerAssignment to be installed. That's why Pip is maintaining v0.0.19. I suggest creating a new conda environment with Python 3.10 and installing the newest SigProfilerAssignment. You can do that by running the following:
```
conda create --name new_SigProfAssignR -y
conda activate new_SigProfAssignR
conda install python=3.10 -y
pip install SigProfilerAssignment
```

After this, you just need to connect the new Python installation with R and Reticulate by using the commands explained in the README. I hope this helps, and please let me know if you still have any issues or have further questions.

Thanks again for your interest! Best,

Marcos

**Comment by eltonjrv** [comment]
Thanks a lot Marcos!
Issue solved under python 3.10.
I've now got the following versions in my new conda environment:
SigProfilerAssignment      0.1.3
SigProfilerMatrixGenerator 1.2.25
sigProfilerPlotting        1.3.20

I'm now experiencing an issue of cosmic_fit not recognizing my MAF format.
I'll try to work around this for a while, and will then open a new issue herein in case I can't circumvent.

Thanks again,
Best,
Elton

**Comment by marcos-diazg** [comment]
Hi Elton!

I'm glad the issue was solved. Regarding any formatting issues, the input handling for all SigProfiler tools is done by [SigProfilerMatrixGenerator](https://github.com/AlexandrovLab/SigProfilerMatrixGenerator) (or its R counterpart, [SigProfilerMatrixGeneratorR](https://github.com/AlexandrovLab/SigProfilerMatrixGeneratorR)). I recommend checking the README of the tools, as well as their [wiki page](https://osf.io/s93d5/wiki/home/). [Example files](https://osf.io/s93d5/wiki/3.%20Using%20the%20Tool%20-%20SBS%2C%20ID%2C%20DBS%20Input/) are usually very useful to troubleshoot any formatting issues with your input files.

I hope this helps, and please feel free to reopen the GitHub ticket if needed.

Best wishes,

Marcos
